acl/authorizer.go

```diff
@@ -161,6 +162,280 @@ type Authorizer interface {
 
 	// Embedded Interface for Consul Enterprise specific ACL enforcement
 	enterpriseAuthorizer
+
+	// ToAllowAuthorizer is needed until we can use ResolveResult in all the places this interface is used.
+	ToAllowAuthorizer() AllowAuthorizer
+}
+
+// AllowAuthorizer is a wrapper to expose the *Allowed methods.
+// This and the ToAllowAuthorizer function exist to tide us over until the ResolveResult struct
+// is moved into acl.
+type AllowAuthorizer struct {
+	Authorizer
+}
+
+// ACLReadAllowed checks for permission to list all the ACLs
+func (a AllowAuthorizer) ACLReadAllowed(ctx *AuthorizerContext) error {
```

Comment:
Followup: do these even need to be methods, or could they be a suite of package functions with a leading argument `Authorizer` so your in-endpoint checks would be:
```
if acl.ACLReadAllowed(authz, ctx) {
  ...
}
```
which is terser than what shows up down below.

Reply:
I'm curious about this too; it feels like added indirection and I think RB's suggested functions read better

Reply:
The intent is to merge AllowAuthorizer and ACLResolveResult from consul/acl.go in a future PR. These would be methods on the joined class